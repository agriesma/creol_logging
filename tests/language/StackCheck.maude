load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "Test" : Class | Inh: noInh, Param: noVid, Att: "depth" |-> null, "max"
      |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: assign( ( "depth" @ "Test" ), ( "max" @ "Test" ) ; int(0)
              :: int(3) ) ; return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "ccglab0" |-> null, "_" |->
              null, Code: call( "ccglab0" ; "this" ; "test" ; emp ) ;
              get( "ccglab0" ; noVid ) ; free( "ccglab0" ) ; return ( emp ) >,
            < "test" : Method | Param: noVid, Att: "ccglab0" |-> null, "temp"
              |-> null, "_" |-> null, Code: assign( "depth" ; "+" ( "depth"
              :: int(1) ) ) ; assign( "temp" ; "depth" ) ; if "<=" ( "depth"
              :: "max" ) th call( "ccglab0" ; "this" ; "test" ; emp ) ;
              get( "ccglab0" ; noVid ) ; free( "ccglab0" ) el skip fi ; await
              "=" ( "depth" :: "temp" ) ; assign( "depth" ; "-" ( "depth" ::
              int(1) ) ) ; return ( emp ) >,
      Ocnt: 0 > } .
endm

