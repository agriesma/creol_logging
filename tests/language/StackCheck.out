==========================================
reduce in PROGRAM : state .
result State: {
< "Test" : Class | Inh: noInh, Param: noVid, Att: "depth" |-> null, "max" |->
    null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assign(("depth" @
    "Test"), ("max" @ "Test") ; int(0) :: int(3)) ; return(emp) >, < "run" :
    Method | Param: noVid, Att: "_" |-> null, "ccglab0" |-> null, Code: call(
    "ccglab0" ; "this" ; "test" ; emp) ; get("ccglab0" ; noVid) ; free(
    "ccglab0") ; return(emp) >, < "test" : Method | Param: noVid, Att: "_" |->
    null, "ccglab0" |-> null, "temp" |-> null, Code: assign("depth" ; "+"(
    "depth" :: int(1))) ; assign("temp" ; "depth") ; if "<="("depth" :: "max")
    th call("ccglab0" ; "this" ; "test" ; emp) ; get("ccglab0" ; noVid) ; free(
    "ccglab0") el skip fi ; await "="("depth" :: "temp") ; assign("depth" ;
    "-"("depth" :: int(1))) ; return(emp) >,
  Ocnt: 0 >
}
Bye.
